backend/src/main/java/com/festago/school/repository/v1/SchoolV1QueryDslRepository.java

```diff
@@ -0,0 +1,132 @@
+        }
+
+        return Optional.of(response.get(0));
+    }
+
+    public SliceResponse<SchoolFestivalV1Response> findFestivalsBySchoolId(Long schoolId, LocalDate today,
+                                                                           SchoolFestivalV1SearchCondition searchCondition) {
+        List<SchoolFestivalV1Response> result =
+            select(new QSchoolFestivalV1Response(festival.id,
+                    festival.name,
+                    festival.startDate,
+                    festival.endDate,
+                    festival.thumbnail,
+                    festivalQueryInfo.artistInfo
+                )
+            )
+                .from(festival)
+                .where(festival.school.id.eq(schoolId),
+                    addPhaseOption(searchCondition.isPast(), today),
+                    addPagingOption(searchCondition.lastFestivalId(), searchCondition.lastStartDate(),
+                        searchCondition.isPast()))
+                .leftJoin(festivalQueryInfo).on(festivalQueryInfo.festivalId.eq(festival.id))
+                .orderBy(addOrderOption(searchCondition.isPast()))
+                .limit(searchCondition.size() + NEXT_PAGE_CHECK_NUMBER)
+                .fetch();
+
+        return createResponse(searchCondition.size(), result);
+    }
+
+    private BooleanExpression addPhaseOption(boolean isPast, LocalDate today) {
+        if (isPast) {
+            return festival.endDate.lt(today);
+        }
+
+        return festival.endDate.goe(today);
+    }
+
+    private BooleanExpression addPagingOption(Long lastFestivalId, LocalDate lastStartDate, boolean isPast) {
+        if (isNotFirstPage(lastFestivalId, lastStartDate)) {
+            if (isPast) {
+                return festival.startDate.lt(lastStartDate)
+                    .or(festival.startDate.eq(lastStartDate)
+                        .and(festival.id.gt(lastFestivalId)));
+            }
+            return festival.startDate.gt(lastStartDate)
+                .or(festival.startDate.eq(lastStartDate)
+                    .and(festival.id.gt(lastFestivalId)));
+        }
+
+        return null;
+    }
+
+    private boolean isNotFirstPage(Long lastFestivalId, LocalDate lastStartDate) {
+        return lastFestivalId != null && lastStartDate != null;
+    }
+
+    private OrderSpecifier<LocalDate>[] addOrderOption(boolean isPast) {
+        if (isPast) {
+            return new OrderSpecifier[]{festival.endDate.desc()};
+        }
+
+        return new OrderSpecifier[]{festival.startDate.asc()};
+    }
+
+    private SliceResponse<SchoolFestivalV1Response> createResponse(
+        Integer pageSize,
+        List<SchoolFestivalV1Response> content) {
+        boolean isLast = true;
+        if (content.size() == pageSize + NEXT_PAGE_CHECK_NUMBER) {
+            content.remove(content.size() - NEXT_PAGE_CHECK_NUMBER);
+            isLast = false;
+        }
+
+        return new SliceResponse(isLast, content);
+    }
```

Comment:
줄바꿈을 다음과 같이 해주는게 어떨까요? 파리미터랑 지역 변수랑 겹쳐서 가독성이 낮아 보이네요!
```suggestion
    private SliceResponse<SchoolFestivalV1Response> createResponse(
        Integer pageSize,
        List<SchoolFestivalV1Response> content
    ) {
        boolean isLast = true;
        if (content.size() == pageSize + NEXT_PAGE_CHECK_NUMBER) {
            content.remove(content.size() - NEXT_PAGE_CHECK_NUMBER);
            isLast = false;
        }

        return new SliceResponse(isLast, content);
    }
```